### PR TITLE
Adding ignore: ["after-declaration"] to declaration-empty-line-before.

### PR DESCRIPTION
**Pull request by heatwaveo8**
Pull request for #1848.

### CODE REVIEW

File: src/rules/declaration-empty-line-before/__tests__/index.js [hunk 1]
```diff
@@ -178,7 +178,7 @@ testRule(rule, {
   }, {
     code: "a {\n/* comment */\ntop: 15px;\n}",
   }, {
-    code: "a {\r\n/* comment */\r\nntop: 15px;\r\n}",
+    code: "a {\r\n/* comment */\r\ntop: 15px;\r\n}",
```
Comment (ntwb) [inline src/rules/declaration-empty-line-before/__tests__/index.js]: Nice catch 👍

**Comment by ntwb** [comment]
Thanks @heatwaveo8 can you also add the new `after-declaration` option to the rules readme please.

We've a document that outlines what's required: http://stylelint.io/developer-guide/rules/#adding-options-to-existing-rules

**Comment by heatwaveo8** [comment]
Sure, can do. A duplicate link was given to me about resources to see for making changes. Maybe this is the one that was missing. Thanks.

**Comment by heatwaveo8** [comment]
Changes are ready.

File: src/rules/declaration-empty-line-before/__tests__/index.js [hunk 2]
```diff
@@ -191,6 +191,37 @@ testRule(rule, {
...
+  }, {
+    code: "\r\ntop: 15px; top: 15px;\r\n",
+  }, {
+    code: "\ntop: 15px;\ntop: 15px;\n",
+  }, {
+    code: "\r\ntop: 15px;\r\ntop: 15px;\r\n",
+  }, {
+    code: "\ntop: 15px;\n\ntop: 15px;\n",
+  }, {
+    code: "\r\ntop: 15px;\r\n\r\ntop: 15px;\r\n",
```
Comment (davidtheclark) [inline src/rules/declaration-empty-line-before/__tests__/index.js]: Please put all these inside rules to better approximate real CSS. Or is there a specific reason you don't have them in rules?

File: src/rules/declaration-empty-line-before/__tests__/index.js [hunk 3]
```diff
@@ -191,6 +191,37 @@ testRule(rule, {
...
+    code: "a {\n top: 15px;\n}",
+    message: messages.expected,
+    line: 2,
+    column: 2,
+  }, {
+    code: "a {\r\n top: 15px;\r\n}",
+    message: messages.expected,
+    line: 2,
+    column: 2,
+  } ],
```
Comment (davidtheclark) [inline src/rules/declaration-empty-line-before/__tests__/index.js]: Please add a couple of reject tests ensuring that declarations after *non-*declarations still trigger the relevant warning.

**Comment by davidtheclark** [comment]
Thanks @heatwaveo8! I had a couple of test comments. Otherwise, LGTM.

File: src/rules/declaration-empty-line-before/index.js [hunk 4]
```diff
@@ -36,6 +36,7 @@ export default function (expectation, options) {
         ignore: [
           "after-comment",
           "inside-single-line-block",
+          "after-declaration",
```
Comment (jeddy3) [inline src/rules/declaration-empty-line-before/index.js]: Minor request, can we get these in alphabetical order please. It's helps when looking through lots of rule code.

**Comment by jeddy3** [comment]
@heatwaveo8 Thanks! I've one additional very minor request.

Other than that, and @davidtheclark's requests, this looks great.

**Comment by heatwaveo8** [comment]
By wrapping it in the anchor tag, all of the rules were failing. I can't quite figure out why... the idea behind it is very similar to after-comment above it.

**Comment by heatwaveo8** [comment]
@davidtheclark By wrapping it in the anchor tag, all of the tests fail. But without rule block it is fine. I can't quite figure out why... the idea is very similar to after-comment above it, so they seem like the right accept tests to me.

**Comment by davidtheclark** [comment]
@heatwaveo8 Any plans to finish this up?

**Comment by heatwaveo8** [comment]
@davidtheclark I have committed a fix for the alphabetical ordering today. I'm trying to resolve the conflict for the CHANGELOG.md file but have now run into a problem where it's trying to commit hundreds of changed files. I'll try and figure out the problem as soon as I can. I'm very new to Git, and it's proving to be a bit of a learning curve... I'm much more familiar with Subversion.

**Comment by heatwaveo8** [comment]
@davidtheclark I have brought in the new version, resolved a CHANGELOG conflict and alphabetized the lines as per jeddy3's request. Will get back to the unit tests in a couple days.

**Comment by heatwaveo8** [comment]
I am done with my changes, but for some reason it is not letting me squash the revisions anymore and seems to have acted up in the process. Would someone be able to help with this?

**Comment by heatwaveo8** [comment]
I am done with my changes, but for some reason it is not letting me squash the revisions anymore and seems to have acted up in the process. Would someone be able to help with this? If not I'll go ahead and create a new fork and pull request. Also, please let me know if I'm doing the new reject rules correctly. Thanks.

**Comment by heatwaveo8** [comment]
I am done with my changes, but for some reason it is not letting me squash the revisions anymore and seems to have acted up in the process. Would someone be able to help with squashing these for me? If not I'll go ahead and create a new fork and pull request. Also, please let me know if I'm doing the new reject rules correctly. Thanks.

**Comment by davidtheclark** [comment]
@heatwaveo8 I'll take a look right now ...

**Comment by davidtheclark** [comment]
Merged in 5710a4a09ee0d3da3954701c5702a8ebcfbcba4b. Thanks @heatwaveo8! 

Not sure what git commands you had tried, but in the future I'd suggest (from your branch) `git rebase master -i`. It's a good command to learn how to use.

**Comment by evilebottnawi** [comment]
@davidtheclark  `git push` without `--force` after `merge`

**Comment by evilebottnawi** [comment]
@heatwaveo8 Right way:
1. https://help.github.com/articles/syncing-a-fork/ (resolve conflicts if there)
2. `git checkout you-branch`
2. `git rebase master`
3. `git push --force`

**Comment by heatwaveo8** [comment]
@davidtheclark No problem, glad to help. Right, that's what I was using originally, only with HEAD.

git rebase -i HEAD~4 (or whatever number I had done).

Once I ran into having to merge new changes from your master branch, though, it wouldn't let me squash them in that manner. So I decided to try the desktop app and it added 4 more pull requests to the mix, why exactly I'm not sure. Anyway I'm glad to see you could bring them in as one commit.

@evilebottnawi Thanks for your advice. Perhaps that's what I needed to do instead. Will try it out that way next time.

**Comment by heatwaveo8** [comment]
@davidtheclark No problem, glad to help. Right, that's what I was using originally, only with HEAD.

git rebase -i HEAD~4 (or whatever number I had done).

Once I ran into having to merge new changes from your master branch, though, it wouldn't let me squash them in that manner. So I decided to try the desktop app and it added 4 more commits to the mix, why exactly I'm not sure. Anyway I'm glad to see you could bring them in as one commit.

@evilebottnawi Thanks for your advice. Perhaps that's what I needed to do instead. Will try it out that way next time.

**Pull request closed without merging**